lib/doorkeeper/models/concerns/revocable.rb

```diff
@@ -8,6 +8,29 @@ def revoke(clock = Time)
       def revoked?
         !!(revoked_at && revoked_at <= Time.now)
       end
+
+      def revoke_in(time)
+        update_attribute :revoked_at, (Time.now + time)
```

Comment:
This reads weird though. The key is written in the past, but the value is in the future. This seems confusing.